src/main/java/seedu/duke/data/timetable/Timetable.java

```diff
@@ -39,17 +58,100 @@ public void setEvents(ArrayList<Event> events) {
      * @param event to be added.
      */
     public void addEvent(Event event) {
-
+        events.add(event);
+        if (!event.getRecurring()) {
+            nonRecurringEvents.add(event);
+        } else {
+            if (event instanceof DailyEvent) {
+                dailyEvents.add((DailyEvent) event);
+            } else if (event instanceof WeeklyEvent) {
+                weeklyEvents.add((WeeklyEvent) event);
+            } else if (event instanceof MonthlyEvent) {
+                monthlyEvents.add((MonthlyEvent) event);
+            } else if (event instanceof YearlyEvent) {
+                yearlyEvents.add((YearlyEvent) event);
+            }
+        }
     }
 
     /**
      * Removes an event from the timetable.
      *
-     * @param event to be removed.
+     * @param index Index to be removed.
```

Comment:
Can include the system exception also
```suggestion
     * @param index Index to be removed.
     * @throws SystemException
```